extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -260,445 +46,49 @@ public function signup() {
-		if ( !( $this->isValidUser( $user ) && $this->isValidSession( $user ) ) ) {
-			return;
-		}
-
-		// check email changes limit
-		$memKey = wfSharedMemcKey( 'wikialogin', 'email_changes', $user->getId() );
-
-		// CONN-471: Respect the registration per email limit
-		if ( !( $this->isWithinEmailChangesLimit( $memKey ) && $this->isWithinRegistrationPerEmailLimit( $email ) ) ) {
-			return;
-		}
-
-		// increase counter for email changes
-		$this->userLoginHelper->incrMemc( $memKey );
-
-		$this->setResponseFields(
-			'ok',
-			wfMessage( 'usersignup-reconfirmation-email-sent', $email )->escaped()
-		);
-		if ( $email != $user->getEmail() ) {
-			$user->setEmail( $email );
-
-			// CONN-471: Call AbortNewAccount to validate username/password with Phalanx
-			if ( $this->isBlockedByPhalanx( $user ) ) {
-				return;
-			}
-
-			// send reconfirmation email
-			$result = $user->sendReConfirmationMail();
-
-			$user->saveSettings();
-
-			// set counter to 1 for confirmation emails sent
-			$memKey = $this->userLoginHelper->getMemKeyConfirmationEmailsSent( $user->getId() );
-			$this->wg->Memc->set( $memKey, 1, 24 * 60 * 60 );
-
-			if ( !$result->isGood() ) {
-				$this->setResponseFields(
-					'error',
-					wfMessage( 'userlogin-error-mail-error', $result->getMessage() )->parse()
-				);
-			}
-		}
+		$this->redirectToAuthPages();
 	}
 
-	/**
-	 * validate form
-	 * @requestParam string field [userloginext01/userloginext02/email/birthdate]
-	 * @requestParam string userloginext01 //CE-413 signup spam attack - changing username field to userloginext01
-	 * @requestParam string email
-	 * @requestParam string userloginext02 //CE-413 signup spam attack - changing password field to userloginext02
-	 * @requestParam string birthmonth
-	 * @requestParam string birthday
-	 * @requestParam string birthyear
-	 * @responseParam string result [ok/error]
-	 * @responseParam string msg - result message
-	 * @responseParam string errParam - error param
-	 */
 	public function formValidation() {
-		$field = $this->request->getVal( 'field', '' );
-		$signupForm = new UserLoginForm( $this->wg->request );
-		$signupForm->load();
+		$this->redirectToAuthPages();
+	}
 
-		switch ( $field ) {
-			case 'userloginext01' :
-				$signupForm->initValidationUsername();
-				break;
-			case 'email' :
-				$signupForm->initValidationEmail()
-					&& $signupForm->initValidationRegsPerEmail();
-				break;
-			case 'birthdate' :
-				$signupForm->initValidationBirthdate();
-				break;
-			default:
-				throw new MWException( "Unhandled case value" );
-		}
 
-		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok';
-		$this->msg = $signupForm->msg;
-		$this->errParam = $signupForm->errParam;
-		if ( $this->request->getVal('retusername') ) {
-			$this->username = $signupForm->mUsername;
-		}
+	private function redirectToAuthPages() {
+		$out = $this->getOutput();
+		$out->redirect( $this->getRedirectUrl() );
+		$this->clearBodyAndSetMaxCache( $out );
```

Comment:
Pending https://github.com/Wikia/app/pull/12817 this code can be extracted to a more common place.